src/main/java/com/vaadin/demo/component/splitlayout/SplitLayoutToggle.java

```diff
@@ -1,21 +1,61 @@
 package com.vaadin.demo.component.splitlayout;
 
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.button.ButtonVariant;
 import com.vaadin.flow.component.html.Div;
-import com.vaadin.flow.component.html.Span;
-import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.component.icon.Icon;
+import com.vaadin.flow.component.icon.VaadinIcon;
 import com.vaadin.flow.component.splitlayout.SplitLayout;
 import com.vaadin.flow.router.Route;
 import com.vaadin.demo.DemoExporter; // hidden-source-line
 
 @Route("split-layout-toggle")
 public class SplitLayoutToggle extends Div {
+    private boolean sidebarCollapsed;
+    private final Button button;
+    private final Div masterContainer;
+    private final Icon leftArrowIcon;
+    private final Icon rightArrowIcon;
+    private final DetailContent detailContent;
 
     public SplitLayoutToggle() {
         // tag::snippet[]
-        SplitLayout splitLayout = new SplitLayout(new Span("Side 1"), new Span("Side 2"));
-        add(splitLayout);
+        button = new Button();
+        masterContainer = new Div();
+        leftArrowIcon = VaadinIcon.ARROW_LEFT.create();
+        rightArrowIcon = VaadinIcon.ARROW_RIGHT.create();
+        detailContent = new DetailContent();
+        MasterContent masterContent = new MasterContent();
+
+        sidebarCollapsed = false;
+
+        button.addClickListener(event -> {
+            sidebarCollapsed = !sidebarCollapsed;
+            setStyles();
+        });
+        button.getElement().setAttribute("aria-label", "Expand/collapse sidebar");
+        button.addThemeVariants(ButtonVariant.LUMO_TERTIARY, ButtonVariant.LUMO_ICON);
+        button.getStyle().set("float", "right");
+
+        masterContainer.add(button, masterContent);
+
+        setStyles();
+
+        SplitLayout splitLayout = new SplitLayout(masterContainer, detailContent);
         // end::snippet[]
+
+        splitLayout.setMaxHeight("280px");
+        add(splitLayout);
+    }
+
+    // tag::setstyles[]
+    private void setStyles() {
+        button.setIcon(sidebarCollapsed ? rightArrowIcon : leftArrowIcon);
+        int sidebarWidthPercentage = sidebarCollapsed ? 13 : 40;
+        masterContainer.getElement().executeJs("this.style = `overflow: hidden; width: ${$0}%`", sidebarWidthPercentage);
+        detailContent.getElement().executeJs("this.style = `width: ${$0}%`",100 - sidebarWidthPercentage);
```

Comment:
There should be a better approach I think. Any idea?

Reply:
Have you tried to change styles via `getStyle().set(...)`? Would it work?

```suggestion
        detailContent.getStyle().set("width", 100 - sidebarWidthPercentage + "%")
```

Reply:
that doesn't work because of the styles added via the client when you change the divider. I need to remove that style and set my own style, but couldn't find a way.

Reply:
I checked the branch locally and I found that the splitter (divider) manages the content width via `flex: 1 1 ...px`. 

Maybe you could use `flex: 1 1 ${percentage}%` rather than `width: ${percentage}%`.

Reply:
I gave it a shot, and it seems to work at first sight.

```java
int sidebarWidthPercentage = sidebarCollapsed ? 13 : 40;

masterContainer.getElement().getStyle().set("flex", "1 1 " + sidebarWidthPercentage + "%");
detailContent.getElement().getStyle().set("flex", "1 1 " + (100 - sidebarWidthPercentage) + "%");
```

Reply:
that works but it diverges from other examples (they're all using `width`) and it kinda relies on the implementation details. IDK. @sissbruecker any idea?

Reply:
The Flow component has a dedicated API for setting the splitter position (sets the size in percentage for the first child):
```
splitLayout.setSplitterPosition(sidebarCollapsed ? 13 : 40);
```

Note that the Typescript example here is flawed: custom splitter positions set by the user will be overriden if there is a change to some other state in the component that triggers a rerender. To solve it, instead of binding the style attribute to an expression, the callback for the button click could manipulate the style on the element.

Reply:
done